clients/unshaded/src/main/java/tachyon/client/file/BaseFileSystem.java

```diff
@@ -0,0 +1,276 @@
+  public void delete(TachyonURI path)
+      throws DirectoryNotEmptyException, FileDoesNotExistException, IOException, TachyonException {
+    delete(path, DeleteOptions.defaults());
+  }
+
+  @Override
+  public void delete(TachyonURI path, DeleteOptions options)
+      throws DirectoryNotEmptyException, FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.delete(path, options);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public boolean exists(TachyonURI path)
+      throws InvalidPathException, IOException, TachyonException {
+    return exists(path, ExistsOptions.defaults());
+  }
+
+  @Override
+  public boolean exists(TachyonURI path, ExistsOptions options)
+      throws InvalidPathException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      // TODO(calvin): Make this more efficient
+      masterClient.getFileInfo(path);
+      return true;
+    } catch (FileDoesNotExistException e) {
+      return false;
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public void free(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    free(path, FreeOptions.defaults());
+  }
+
+  @Override
+  public void free(TachyonURI path, FreeOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.free(path, options);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public URIStatus getStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    return getStatus(path, GetStatusOptions.defaults());
+  }
+
+  @Override
+  public URIStatus getStatus(TachyonURI path, GetStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      return new URIStatus(masterClient.getFileInfo(path));
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public List<URIStatus> listStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    return listStatus(path, ListStatusOptions.defaults());
+  }
+
+  @Override
+  public List<URIStatus> listStatus(TachyonURI path, ListStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    List<FileInfo> fileInfos;
```

Comment:
I prefer just appending `s` which makes it easier to change the type of collection if need be.